libs/awscdk/src/api.ts

```diff
@@ -49,17 +49,14 @@ export class Api extends cloud.Api implements IAwsApi {
     method: string,
     path: string,
     inflight: cloud.IApiEndpointHandler,
-    props?: cloud.ApiGetOptions
+    props?: cloud.FunctionProps,
```

Comment:
I think this should still be `ApiGetOptions`, but it should extend `FunctionProps`. Otherwise, if we want to add any options it will be a breaking change.